dockstore-webservice/src/main/java/io/dockstore/webservice/core/metrics/Metric.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2024 OICR and UCSC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *           http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package io.dockstore.webservice.core.metrics;
+
+import io.swagger.annotations.ApiModelProperty;
+import io.swagger.v3.oas.annotations.media.Schema;
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.Inheritance;
+import jakarta.persistence.InheritanceType;
+import jakarta.persistence.SequenceGenerator;
+
+@Entity
+@Inheritance(strategy = InheritanceType.TABLE_PER_CLASS)
+@Schema(name = "Metric", description = "Describes an aggregated metric", subTypes = { ExecutionTimeStatisticMetric.class, CpuStatisticMetric.class, MemoryStatisticMetric.class, CostStatisticMetric.class, ExecutionStatusCountMetric.class, ValidationStatusCountMetric.class })
```

Comment:
I have mixed feelings about this, openapi inheritance is definitely weird

Reply:
I primarily did it so I only had to set the number of skipped executions in [one place](https://github.com/dockstore/dockstore-support/blob/cdccbb5a563d2850487aa3aae2bacd0c1b03ac62/metricsaggregator/src/main/java/io/dockstore/metricsaggregator/helper/ExecutionAggregator.java#L93) instead of repeating it in each aggregator class.

It also allowed me to bound the generic `M` type https://github.com/dockstore/dockstore-support/blob/cdccbb5a563d2850487aa3aae2bacd0c1b03ac62/metricsaggregator/src/main/java/io/dockstore/metricsaggregator/helper/ExecutionAggregator.java#L19